src/index.js

```diff
@@ -45,13 +48,213 @@ module.exports = {
       }],
     }, {
       type: 'function',
-      fn(opts) { return localRunner.run(opts); },
+      fn(opts) { 
+      // if W doesn't merge, return run(), if W merge, return statistic()
+//        console.log('opts.remoteResult.data is',opts.remoteResult.data);
+//         console.log('opts.previousData is', opts.previousData);
+       console.log('opts.remoteResult.data.endOfIteration is' , opts.remoteResult.data.endOfIteration);
+       console.log('opts.remoteResult.data.statisticStep is', opts.remoteResult.data.statisticStep);
+        
+        if ((opts.remoteResult.data.endOfIteration == true) && (opts.remoteResult.data.statisticStep == 0)){
```

Comment:
Yes, we could merge this part into the local.js, or we can have local_comp1.js, local_comp2.js....... and on the remote node, we can also have remote_comp1.js, local_comp2.js..... ,  maybe you can help me on this. I did plan to write separate code for each condition.......Probably show me an example of modularizing code, I think single shot also can be modularized